adapter/ryu/omniui/omniui.py

```diff
@@ -100,6 +101,14 @@ def switch_features_handler(self, ev):
                                           ofproto.OFPCML_NO_BUFFER)]
         self.add_flow(datapath, 0, match, actions)
 
+    def add_flow10(self, datapath, priority, match, actions):
+        ofproto = datapath.ofproto
+        parser = datapath.ofproto_parser
+
+        mod = parser.OFPFlowMod(datapath=datapath, match=match, cookie=0, command=ofproto.OFPFC_ADD, idle_timeout=0,
+                                hard_timeout=0, priority=priority, flags=ofproto.OFPFF_SEND_FLOW_REM, actions=actions)
+        datapath.send_msg(mod)
+
```

Comment:
It seems that the flow entries installed by `omniui.py` are slightly different from the ones installed by `simple_switch.py`. This situation happens on both OpenFlow10 & OpenFlow13. I still recommend you to try to split two functionalities apart (learning switch mechanism and event forwarding). Let the original `simple_switch.py` and `simple_switch_13.py` do their jobs, unless the same event cannot handled by two or more functions at the same time.